deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobIT.java

```diff
@@ -476,6 +476,50 @@ public void resumeCompletedWorkIngestTest() throws Exception {
         ingestedObjectsCount(3);
     }
 
+    @Test
+    public void ingestWorkObjectObjCountWithRetry() throws Exception {
+        String label = "testwork";
+        PID workPid = pidMinter.mintContentPid();
+
+        // Construct the deposit model with work object
+        Model model = job.getWritableModel();
+        Bag depBag = model.createBag(depositPid.getRepositoryPath());
+
+        // Constructing the work in the deposit model with a label
+        Bag workBag = model.createBag(workPid.getRepositoryPath());
+        workBag.addProperty(RDF.type, Cdr.Work);
+        workBag.addProperty(CdrDeposit.label, label);
+
+        PID mainPid = addFileObject(workBag, FILE1_LOC, FILE1_MIMETYPE, FILE1_SHA1, FILE1_MD5);
+        PID mainPid2 = addFileObject(workBag, FILE1_LOC, FILE1_MIMETYPE, null, FILE1_MD5_BAD);
+
+        depBag.add(workBag);
+
+        workBag.asResource().addProperty(Cdr.primaryObject,
+                model.getResource(mainPid.getRepositoryPath()));
+
+        workBag.asResource().addProperty(Cdr.primaryObject,
```

Comment:
there should only be one primaryObject per work (that's enforced elsewhere). Also, you don't need the `.asResource()` part